app/controllers/concerns/client_sortable.rb

```diff
@@ -24,38 +24,52 @@ def filtered_and_sorted_clients(default_order: nil)
 
   # see if there are any overlapping keys in the provided params and search/sort set
   def has_search_and_sort_params?
-    (params.keys.map(&:to_sym) & search_and_sort_params).any?
+    overlapping_keys = (params.keys.map(&:to_sym) & search_and_sort_params)
+    hash_params = params.try(:to_unsafe_h) || params
+    overlapping_keys.any? && hash_params.slice(*overlapping_keys).any? { |_, v| v.present? }
   end
 
   private
 
   def setup_sortable_client
-    reset_filter_params if params[:clear]
+    delete_cookie if params[:clear]
     @sort_column = clients_sort_column
     @sort_order = clients_sort_order
-    @filters = {
-      search: normalize_phone_number_if_present(params[:search]),
-      status: status_filter,
-      stage: stage_filter,
-      assigned_to_me: params[:assigned_to_me],
-      unassigned: params[:unassigned],
-      needs_response: params[:needs_response],
-      year: params[:year],
-      vita_partner_id: params[:vita_partner_id].present? ? params[:vita_partner_id].to_i : nil,
-      assigned_user_id: params[:assigned_user_id].present? ? params[:assigned_user_id].to_i : nil,
+    filter_source = has_search_and_sort_params? ? params : cookie_filters
+    @filters = filters_from(filter_source)
+    set_cookie
```

Comment:
This is implemented using cookies, but I'd prefer to see it implemented using our existing session. This has a few advantages in my opinion:

* The session is cleared when users log out. This means that if two users use the same device after each other, their filters won't persist across users. Mostly this is important to me aesthetically! It does have a minor practical implication - for example, if the first user searches for a name/email and then the second user visits the same area in the site, the second user will see a name/email they weren't supposed to know was a client.

* The session uses Rails for checking authenticity (it's stored with signed & encrypted cookies). There's nothing I see here that really requires that level of authenticity checking, but it's nice to lean on when it's there imho.

I know that leaning on the session means that the data will get lost if the Hub user is inactive for 30 minutes. I think that's OK for now - I _think_ that most of the need here is when people click a client from a filtered Clients view, then see the same filter a few minutes later when navigating around the site.

Related - if that is the only need, do you think that adjusting the `back_to` behavior would meet the user need, too? That is, when you click on a Client, there's a "<< back to All Clients" link at the top of the Client page; that could be adjusted to include the query params of the filter that let the user navigate to this client. It would be more work in some ways, but I do personally being explicit about state management. It would also ensure one browser tab can't interfere with another; if you're trying to do two filters in two different browser tabs, the session/cookie approach will result in them clobbering each other, which I personally don't like. (Not a blocker.)

Another option is to consider storing these filters in the database. If our users really want this to persist, then maybe that's better!

If we want the transient approach, I do have a strong preference for session over cookie here. Let me know what you think.

Reply:
The session cookie has a limited length, so you're at real risk of cramming too much into the session if we were to save a hash of filters for each dashboard into the filters -- I believe based on code I looked at that this might have happened last season in the session cookie on the intake side of things? 

For this reason, I think its important to be super discerning of what's tied to the session and what can be managed using a regular cookie. I think the risks here off overloading the session cookie outweigh some of the funkiness with storing in a cookie. 

I do think that a better way to do this would be to use localstorage and do this in JS, but based on the current implementation of the filters a cookie was more straightforward -- we can set cookies on the server, but can only set/retrieve localstorage in js. sometimes you have to choose the most optimal solution based on the time constraints and limitations of the original implementation, so using a cookie felt more straightforward than figuring out the best way/place to save into localStorage.

I think this filtering and sorting logic is a really high-touch area of the code and will likely continue to be refactored based on changing needs, so my thought is to get something out the door and then we revise based on ongoing feedback.